### PR TITLE
Enhance streaming_compression examples.

### PR DESCRIPTION
**Pull request by marxin**
Add level argument to the first test and be more verbose about
used compression level and number of threads.

### CODE REVIEW

File: examples/streaming_compression.c [hunk 1]
```diff
@@ -15,9 +15,12 @@
 #include <zstd.h>      // presumes zstd library is installed
 #include "common.h"    // Helper functions, CHECK(), and CHECK_ZSTD()
 
+#define THREADS 4
```
Comment (senhuang42) [inline examples/streaming_compression.c]: I don't think a macro is necessary here. You can pass a `const int nbThreads = 4` from `main()` and add an `nbThreads` arg to `compressFile_orDie`. That way it's more consistent with the change to `cLevel` getting passed from main as well (and in fact, you could add `nbThreads` as an additional arg to the CLI too).

File: examples/streaming_compression.c [hunk 2]
```diff
@@ -106,17 +109,22 @@ int main(int argc, const char** argv)
 {
     const char* const exeName = argv[0];
 
-    if (argc!=2) {
+    if (argc!=3) {
```
Comment (senhuang42) [inline examples/streaming_compression.c]: Nit: I'd prefer if we were able to keep the usage as `exeName FILE` with the optionality of specifying `exeName FILE LEVEL` and maybe even `exeName FILE LEVEL NUM_THREADS`, and default back to `cLevel = 1` and `nbThreads=4` or something.

File: examples/streaming_compression_thread_pool.c [hunk 3]
```diff
@@ -56,7 +58,7 @@ static void *compressFile_orDie(void *data)
      */
     CHECK_ZSTD( ZSTD_CCtx_setParameter(cctx, ZSTD_c_compressionLevel, args->cLevel) );
     CHECK_ZSTD( ZSTD_CCtx_setParameter(cctx, ZSTD_c_checksumFlag, 1) );
-    ZSTD_CCtx_setParameter(cctx, ZSTD_c_nbWorkers, 16);
+    ZSTD_CCtx_setParameter(cctx, ZSTD_c_nbWorkers, THREADS);
```
Comment (senhuang42) [inline examples/streaming_compression_thread_pool.c]: Nit: ditto for the unnecessary macro - prefer using local `const` variable.

**Review by senhuang42 — CHANGES_REQUESTED** [review]
Overall I like the improvement to the example! Just some minor suggestions.

**Comment by marxin** [comment]
Thank you for the comments, I made the suggested changes. I like your suggestions.

**Comment by marxin** [comment]
@senhuang42 May I please ping this?

**Review by senhuang42 — APPROVED** [review]
lgtm, thanks for the update!

**Comment by senhuang42** [comment]
I see some tests are failing, can you update the base repo on top of which this branch sits to the latest dev branch and then force push onto this PR again?

**Comment by marxin** [comment]
Sure, I've just done that.

**Review by Cyan4973 — APPROVED** [review]

**Pull request merged**